graphFunctions.js

```diff
@@ -0,0 +1,296 @@
+// Function to reorganize the data to limit parameter constraints
+// Inputs: data, an Array of sub-arrays (defined below),
+//			limit, Integer value of maximum allowed devices,
+//			setting, Integer value depending on structure of input
+//			data
+function reduceLength(data, limit, setting){
+	var newData = [];
+	// Make sure the data needs to be modified
+	if (data.length > limit){
+		var sum = 0;
+		// determine input data structure
+		// setting === 0, data is in [Name, value] format
+		// setting === 1, data is in [Rank, Name, value, percentage] format
+		if (setting <= 1){
+			for (var i = 0; i < limit; i++){
+				// Consider using a splice here since the structure
+				// remains the same but for loop still needed for
+				// sum
+				var value = parseInt(data[i][1 + setting]);
+				newData.push([data[i][0 + setting],value]);

```

Comment:
This doesn't do any data shifting. It simply reduces the data points to the desired length. So if the setting is 20, it takes the first 20 variables then sums up the remainder. I considered using an array splice but since the data has to be restructured if the setting === 0, I didn't see a real point to doing it given the maximum number of data points is 100.